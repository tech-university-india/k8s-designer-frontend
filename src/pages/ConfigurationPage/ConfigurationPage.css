body {
    background-color: white;
    
}
.config-body{
    border: 1px solid rgb(82, 81, 81);
    height: 98vh;
    width: 24.25vw;
    background-color: white;
    margin: 0 0 0 75.4vw;
    font-family: 'Open Sans', sans-serif;
    font-weight: 800;
    overflow: auto;
}
.config-element-border{
    height: 5vh;
    font-size: 20px;
    border-bottom : 2px solid rgb(82, 81, 81);
    text-align: center;
    padding-top: 2vh;
    
}

@media screen and (max-width:768px) {
    .config-element-border{
        font-size: 15px;
        height: 5vh;
        padding-top: 1.5vh;

    }
    
}