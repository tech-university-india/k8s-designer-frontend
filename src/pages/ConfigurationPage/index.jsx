import './ConfigurationPage.css';
import * as React from 'react';
import Button from '@mui/material/Button';
import {Header,Configuration} from '../../components';

export default function ConfigurationPage() {

  const [service,setService] = React.useState();

  const handleSelectService =(selectedService)=>{
    setService(selectedService);
  };

  return (
    <div>
      <Header /> 
      <Button className='config-btn-color' onClick={()=>{handleSelectService('FRONTEND');}}>FRONTEND</Button>
      <Button className='config-btn-color' onClick={()=>{handleSelectService('BACKEND');}}>BACKEND</Button>
      <Button className='config-btn-color' onClick={()=>{handleSelectService('DATABASE');}}>DATABASE</Button>

      <div className='config-body'>
        <div className='config-element-border'>CONFIGURATION</div>
        {service !==undefined &&  <Configuration service = {service} />}
      </div>
    </div>
    
  );
}