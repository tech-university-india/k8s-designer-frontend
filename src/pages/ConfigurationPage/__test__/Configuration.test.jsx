import {fireEvent, render ,screen } from '@testing-library/react';
import ConfigurationPage from '../index';
import * as React from 'react';


describe('Configuration',()=>{

  it('should show configuration text when component is rendered',()=>{
    expect(screen.queryByText('CONFIGURATION')).toBeFalsy();
    render(<ConfigurationPage/>);
    expect(screen.getByText('CONFIGURATION')).toBeTruthy();
  }); 
  it('should render Backend configuration component when rendered with service value equal to backend',()=>{
    render(<ConfigurationPage/>);
    const BackendButton = screen.getByText('BACKEND');
    fireEvent.click(BackendButton);
    expect(screen.queryByText('BACKEND MICROSERVICES')).toBeTruthy();
  });  
  it('should render Database configuration component when rendered with service value equal to database',()=>{
    render(<ConfigurationPage/>);
    const DatabaseButton = screen.getByText('DATABASE');
    fireEvent.click(DatabaseButton);
    expect(screen.queryByText('DATABASE MICROSERVICES')).toBeTruthy();
  });  
  it('should render Frontend configuration component when rendered with service value equal to frontend',()=>{
    render(<ConfigurationPage/>);
    const FrontendButton = screen.getByText('FRONTEND');
    fireEvent.click(FrontendButton);
    expect(screen.queryByText('FRONTEND MICROSERVICES')).toBeTruthy();
  });  
  it('should render correctly and create a snapshot', () => {
    const {asFragment } =  render(<ConfigurationPage/>);
    expect(asFragment()).toMatchSnapshot();
  });
});