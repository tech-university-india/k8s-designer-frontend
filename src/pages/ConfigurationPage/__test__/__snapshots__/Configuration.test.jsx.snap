// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Configuration should render correctly and create a snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="header"
    >
      <div
        class="logo"
      >
        K8D
      </div>
      <div
        class="user"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault avatar css-2s90m6-MuiAvatar-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
            data-testid="PersonIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
            />
          </svg>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Logout
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium config-btn-color css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      FRONTEND
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium config-btn-color css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      BACKEND
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium config-btn-color css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      DATABASE
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <div
      class="config-body"
    >
      <div
        class="config-element-border"
      >
        CONFIGURATION
      </div>
    </div>
  </div>
</DocumentFragment>
`;
